include/tracing/tracing.h

```diff
@@ -131,6 +135,41 @@
  */
 #define sys_trace_idle()
 
+/**
+ * @brief Trace initialisation of a Semaphore
+ * @param sem Semaphore object
+ */
+#define sys_trace_semaphore_init(sem)
+
+/**
+ * @brief Trace taking a Semaphore
+ * @param sem Semaphore object
+ */
+#define sys_trace_semaphore_take(sem)
```

Comment:
why are these preprocessor `#defines`?

these should be empty inline functions for type safety

never use a macro when an inline function will suffice IMO

Reply:
if I remember correctly we used to have them inline initially or at least in the  original PR from 2-3 years ago, some users however wanted them to be defines to make it easy to integrate with existing tracing formats. There was some other complication I do not recall, need to check and will revisit, in any case, this would need to happen in a different PR.